.github/workflows/build_all_apps.yml

```diff
@@ -3,313 +3,91 @@ name: Build all apps on latest SDK
-\"app-firmachain\",
-\"app-coti\",
-\"app-phala\",
-\"app-bitcoin\",
-\"app-bifrost-kusama\",
-\"app-flow\",
-\"app-bifrost-new\",
-\"app-bifrost\",
-\"app-constellation\",
-\"app-freeton\",
-\"app-aergo\",
-\"app-parallel\",
-\"app-thorchain\",
-\"app-centrifuge\",
-\"app-passwords\",
-\"app-nodle\",
-\"app-oxen\",
-\"app-navcoin\",
-\"app-astar\",
-\"app-helium\",
-\"app-casper\",
-\"app-shimmer\",
-\"app-plugin-compound\",
-\"app-polkadex\",
-\"app-xxnetwork\",
-\"app-decimal\",
-\"app-axelar\",
-\"app-aeternity\",
-\"app-secret\",
-\"app-desmos\",
-\"app-medibloc\",
-\"app-persistence\",
-\"app-platon\",
-\"app-elastos\",
-\"app-groestlcoin\",
-\"app-iota-legacy\",
-\"app-mimblewimblecoin\",
-\"app-genshiro\",
-\"app-particl\",
-\"app-ont\",
-\"app-harmony-one\",
-\"app-iov\",
-\"app-windows-hello\",
-\"app-hathor\",
-\"app-safecoin\",
-\"app-handshake\",
-\"app-conflux\",
-\"app-theta\",
-\"app-tendermint\",
-\"app-sia-x\",
-\"app-qrl\",
-\"app-nano\",
-\"app-nimiq\",
-\"app-kin-spl\",
-\"app-kin\",
-\"app-icx\",
-\"app-hycon\",
-\"app-fic\",
-\"app-factom\",
-\"app-binance\",
-\"app-ark\",
-\"app-ardor\",
-\"app-aion\",
-\"app-neo3\",
-]}
-'
-# Failing apps
-#\"app-avalanche\",
-#\"app-kadena\",
-#\"app-security-key\",
-#\"app-starknet\",
-#\"app-panacea\",
-#\"app-insolar\",
-#\"app-dgld\",
-#\"app-sia\",
-#\"app-streaming\",
-#\"app-xinfin\",
-#\"app-woleet\",
-#\"app-reef\",
-#\"app-edgeware\",
-#\"app-celo-validator-signer\"
-#\"app-hodl\",
-#\"app-equilibrium\",
-#\"app-revealer\",
-#\"app-rise\",
-#\"app-keepass\",
-#\"app-bitsong\",
-
 jobs:
   prepare-matrix:
     runs-on: ubuntu-latest
     outputs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
-      - id: set-matrix
-        run: echo "matrix=${{  env.repo_list  }}" >> $GITHUB_OUTPUT
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install tomlq
+        run: |
+          python -m pip install tomlq
+
+      - name: Get C apps repos
+        run: |
+          page=1
+          result=()
+          while true; do
+            repos=$(curl -s "https://api.github.com/orgs/LedgerHQ/repos?page=$page&per_page=200")
+
+            if ! echo "$repos" | jq -e '.[].name' >/dev/null; then
+              # all repos have been listed
+              echo "all repos have been listed"
+              break
+            fi
+
+            # add filtered repos to the list
+            for repo in $(echo "$repos" | jq -r '.[] | select((.name | startswith("app-")) and (.archived == false) and (.private == false)) | .name'); do
+              result+=("$repo")
+            done
+
+            ((page++))
+          done
+
+          echo "repo_list=${result[@]}" >> $GITHUB_ENV
+
+      - name: Extract metadata
+        id: set-matrix
+        run: |
+          repo_info="["
+          for repo in ${{ env.repo_list }}; do
+            toml_content=$(curl -s "https://raw.githubusercontent.com/LedgerHQ/$repo/develop/ledger_app.toml")
```

Comment:
Is there a way to not explicitly use `develop` here?
Or this should at least be also enforced in `Clone App` step below as `develop` might not be the default branch.

Reply:
@xchapron-ledger I slightly modified the CI workflow by first retrieving the default branch, then using this branch to fetch the manifest, and finally cloning and building from it.